modules/pam_limits/pam_limits.c

```diff
@@ -487,6 +487,46 @@ static int init_limits(pam_handle_t *pamh, struct pam_limit_s *pl, int ctrl)
     return retval;
 }
 
+/*
+ * Read the contents of /proc/sys/fs/<name>
+ * return 1 if conversion succeeds, result is in *valuep
+ * return 0 if conversion fails.
+ */
+static int
+value_from_proc_sys_fs(const char *name, rlim_t *valuep)
+{
+    char pathname[128];
+    char buf[128];
+    FILE *fp;
+    int retval;
+
+    retval = 0;
+
+    snprintf(pathname, sizeof(pathname), "/proc/sys/fs/%s", name);
```

Comment:
I don't see anything specific to /proc/sys/fs/ here.
What if you pass the whole pathname to this function?

Reply:
No problem here. The current construct is a relic from an intermediate idea to allow filenames as "values", ie
```
* - nofile nr_open
```